extensions/imdb/src/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { distance } from "fastest-levenshtein";
+import { List } from "@raycast/api";
+import { useState, useEffect } from "react";
+
+import type { Title } from "./types";
+
+import { getTitle } from "./api";
+import { ListItem } from "./components/ListItem";
+import { debounce } from "./utils";
+
+export default function SearchResults() {
+  const [state, setState] = useState<{ titles: Title[] }>({ titles: [] });
+  const [loading, setLoading] = useState<boolean>(false);
+  const [search, setSearch] = useState<string>("");
+
+  useEffect(() => {
+    async function fetch() {
+      const result = await getTitle(search);
+
+      if (result !== null) {
+        // only add new results
+        if (!state.titles.map((title) => title.imdbID).includes(result.imdbID)) {
+          setState((previous) => ({
+            titles: [result, ...previous.titles],
+          }));
+        }
+      }
+      setLoading(false);
+    }
+    // if a user is still typing, delay the search
+    debounce(fetch, 500);
```

Comment:
Custom throttle should not be used, Raycast have support for [throttle built-in](https://developers.raycast.com/api-reference/user-interface/list#props)